src/agreementManager/ForkService.ts

```diff
@@ -73,67 +84,108 @@ export default class ForkService {
                 "AgreementManager - addBlock - forkCnt is not correct"
             );
 
-        const agreement = this.agreementByBlock(block);
+        const agreement = this.getAgreementByBlock(
+            EvmUtils.decodeBlock(signedBlock.encodedBlock)
+        );
         if (agreement)
             // this should never happen since checks are done before
             throw new Error(
                 "AgreementManager - addBlock - double sign or incorrect data"
             );
-
+        const blockConfirmation: BlockConfirmationStruct = {
+            signedBlock: signedBlock,
+            signatures: []
+        };
         this.addAgreement(forkCnt, {
-            block,
-            blockSignatures: [originalSignature],
-            encodedState
+            blockConfirmation,
+            encodedState,
+            addressesInThreshold: snapShot.participants,
+            snapShot
         });
     }
 
     /**
      * Adds a transaction record to the chainBlocks array for a specific fork
      */
     public addChainBlock(
-        forkCnt: number,
-        transactionCnt: number,
-        participantAdr: string,
+        signedBlock: SignedBlockStruct,
         timestamp: number
     ): void {
+        const { forkCnt } = BlockUtils.getCoordinates(
+            EvmUtils.decodeBlock(signedBlock.encodedBlock)
+        );
         if (!this.isValidForkCnt(forkCnt)) {
             throw new Error("ForkService - addChainBlock - Invalid fork count");
         }
 
         this.forks[forkCnt].chainBlocks.push({
-            transactionCnt,
-            participantAdr,
+            signedBlock,
             timestamp
         });
     }
 
     /*────────── getters ──────────*/
+
+    collectMilestoneSnapshots(forkCnt: number): StateSnapshotStruct[] {
+        const snapShotCommitments: { commitment: BytesLike; height: number }[] =
+            [];
+        const stateSnapshots = [];
+        for (
+            let i = 0;
+            i < this.forks[forkCnt].forkProof.forkMilestoneProofs.length;
+            i++
+        ) {
+            const blockDecoded = EvmUtils.decodeBlock(
+                this.forks[forkCnt].forkProof.forkMilestoneProofs[i]
+                    .blockConfirmations[0].signedBlock.encodedBlock
+            );
+            snapShotCommitments.push({
+                commitment: blockDecoded.stateSnapshotHash,
+                height: Number(blockDecoded.transaction.header.transactionCnt)
+            });
+        }
+        for (const snapShotCommitment of snapShotCommitments) {
+            const stateSnapshot =
+                this.forks[forkCnt].agreements[snapShotCommitment.height]
+                    .snapShot;
+            if (
+                ethers.keccak256(
+                    EvmUtils.encodeStateSnapshot(stateSnapshot)
+                ) === snapShotCommitment.commitment
+            ) {
+                stateSnapshots.push(stateSnapshot);
+            }
+        }
+        return stateSnapshots;
+    }
```

Comment:
I understand what you wanted to do here, but don't understand the rational behind the steps taken to implement it. This can all be done within the single for loop, but more importantly - forkProof.forkMilestoneProofs never gets populated so not sure on what this is running. The function collectMilestoneSnapshots is also not used anywhere. I do understand when you leave a 'TODO' that something will be implemented when you need to use it, but don't have it, but I don't understand the rational of implementing stuff that is not used